server/src/main/java/org/elasticsearch/search/aggregations/timeseries/TimeSeriesIndexSearcher.java

```diff
@@ -78,7 +78,7 @@ public void search(Query query, BucketCollector bucketCollector) throws IOExcept
         PriorityQueue<LeafWalker> queue = new PriorityQueue<>(searcher.getIndexReader().leaves().size()) {
             @Override
             protected boolean lessThan(LeafWalker a, LeafWalker b) {
-                return a.timestamp < b.timestamp;
+                return a.timestamp > b.timestamp;
```

Comment:
I have changed the sorting order because I was getting results in a wrong timestamp order when receiving documents from different segments. By changing this, the doc order problem was fixed. 

I think this bug is fixed by this PR: https://github.com/elastic/elasticsearch/pull/85526

Reply:
Issue fixed by PR https://github.com/elastic/elasticsearch/pull/85526